### PR TITLE
fix serde issue with desub

### PR DESCRIPTION
**Pull request by insipx**
Fix an issue I came across when using with Archive

there is an issue when serializing `to_value` and the `#[serde(flatten)]` attribute with `u128` values.

This PR removes the use of `flatten` from json-resolver deserialization, and enables `arbitrary_precision` feature for the facade crate

### CODE REVIEW

**Review by jsdw — APPROVED** [review]
LGTM!

I'm always a bit scared pointing at master for things because it could break at any time relative to some tag/stable branch, but I guess at the same time  it aligns with what other libraries point to and so is good from a resolving point of view?

**Comment by insipx** [comment]
> LGTM!
> 
> I'm always a bit scared pointing at master for things because it could break at any time relative to some tag/stable branch, but I guess at the same time it aligns with what other libraries point to and so is good from a resolving point of view?

I actually meant to bring this up in conversation at some point

I was thinking we could point desub master at substrate master, and then release with specific tags (like archive)

**Comment by jsdw** [comment]
That sounds good to me I guess that the release process would be to create a release branch, update all substrate refs in the `desub*` crates to eg `polkadot-9.12` and then publish those crates (remembering to update the version on master, too)?

File: desub-current/Cargo.toml [hunk 1]
```diff
@@ -8,8 +8,8 @@ edition = "2018"
 [dependencies]
 log = "0.4"
 thiserror = "1.0.30"
-sp_core = { package = "sp-core",  git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.12" }
-sp_runtime = { package = "sp-runtime", git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.12" }
+sp_core = { package = "sp-core",  git = "https://github.com/paritytech/substrate", branch = "master" }
```
Comment (niklasad1) [inline desub-current/Cargo.toml:11]: would be good pin to a certain commit or are we relying on `Cargo.lock` for that?

File: desub/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ desub-json-resolver = { path = "../desub-json-resolver/", optional = true }
 thiserror = "1.0.30"
 frame-metadata = "14.2"
 codec = { version = "2.3.1", package = "parity-scale-codec" }
-serde_json = "1.0.68"
+serde_json = { version = "1.0", features = ["preserve_order", "arbitrary_precision"] }
```
Comment (niklasad1) [inline desub/Cargo.toml:18]: is `preserve_order` required to work properly?

I just read it was an issue with `arbitary_precision`....

**Review by niklasad1 — APPROVED** [review]

**Review by dvdplm — APPROVED** [review]

**Comment by insipx** [comment]
> That sounds good to me I guess that the release process would be to create a release branch, update all substrate refs in the `desub*` crates to eg `polkadot-9.12` and then publish those crates (remembering to update the version on master, too)?

yeah, it would have to be a pass with `diener`. we could even do a monthly release to follow substrate's monthly releases, but at minimum we should follow the polkadot releases

**Pull request merged**